server/plugin/client.go

```diff
@@ -139,8 +139,14 @@ func (c *client) callFormURLEncoded(url, path, method string, out interface{}, f
 func (c *client) CreateSubscription(body *serializers.CreateSubscriptionRequestPayload, project *serializers.ProjectDetails, channelID, pluginURL, mattermostUserID string) (*serializers.SubscriptionValue, int, error) {
 	subscriptionURL := fmt.Sprintf(constants.CreateSubscription, body.Organization)
 
+	encryptedWebhookSecret, err := c.plugin.Encrypt([]byte(c.plugin.getConfiguration().WebhookSecret), []byte(c.plugin.getConfiguration().EncryptionSecret))
```

Comment:
What was the need for encrypting the webhook secret?

Reply:
1. on Azure we can get this secret only as query-param and when special characters like `+, -, etc` is present then query param breaks so it's better to send and receive an encoded string
1. encryption is done from the security checks perspective if someone steals the webhook token from the Azure portal then that can be misused to post on MM